src/openrct2/windows/GameBottomToolbar.cpp

```diff
@@ -603,6 +641,38 @@ static void window_game_bottom_toolbar_draw_news_item(rct_drawpixelinfo *dpi, rc
     }
 }
 
+static void window_game_bottom_toolbar_draw_middle_panel(rct_drawpixelinfo *dpi, rct_window *w)
+{
+    rct_widget *middleOutsetWidget = &window_game_bottom_toolbar_widgets[WIDX_MIDDLE_OUTSET];
+
+    gfx_fill_rect_inset(
+        dpi,
+        w->x + middleOutsetWidget->left + 1,
+        w->y + middleOutsetWidget->top + 1,
+        w->x + middleOutsetWidget->right - 1,
+        w->y + middleOutsetWidget->bottom - 1,
+        w->colours[1],
+        INSET_RECT_F_30
+    );
+
+    sint32 x = w->x + (middleOutsetWidget->left + middleOutsetWidget->right) / 2;
+    sint32 y = w->y + middleOutsetWidget->top + 11;
+    sint32 width = middleOutsetWidget->right - middleOutsetWidget->left - 62;
+
+    // Check if there is a map tooltip to draw
+    rct_string_id stringId;
+    memcpy(&stringId, gMapTooltipFormatArgs, sizeof(rct_string_id));
```

Comment:
`gMapTooltipFormatArgs` is of a different type than rct_string_id (`uint8[]` vs `uint16` iirc). This copies the correct bits directly to stringId with a single statement. This line is the same as in the other map tooltip code. If taking multiple individual parts from `gMapTooltipFormatArgs` and bitshifting them together is preferred, I'll happily change both.